distributions/aws/examples/storage/efs/auto-efs-setup.py

```diff
@@ -553,7 +562,7 @@ def footer():
     help='Your cluster name (eg: mycluster-1)',
     required=True
 )
-EFS_FILE_SYSTEM_NAME_DEFAULT = "KubeflowEfs"
+EFS_FILE_SYSTEM_NAME_DEFAULT = rand_name("Kubeflow-efs")
```

Comment:
Could I get more context on this change as it seems to change the behaviour of the script?   

The default behaviour ensured that if the script was repeated it would not recreate resources (same behaviour for RDS & S3 script).  
The user would have to specify a new EFS file system name for a new file system to be created. The idea is that with a new file system might come a different configuration which is very hard to guess. For instance, will the user want to keep the same security group? I think It's harder to always be right when we take a stance like this.  
  
That being said, If we think always creating a new EFS file system and re-using the security group is probably the best default then that's fine as the user can also decide to pass a hardcoded EFS file name via the `--efs_file_system_name` parameter to prevent creating a new one.  
  
If we decide to go with this behaviour then we must take the following into account :  
- The RDS & S3 script behaviour would be different (the RDS & S3 script does not recreate a RDS and S3 bucket on every run of the script, to do so would require to generate a random RDS db instance name & random RDS secret name. It's doable but we just gotta keep that in mind.
- We should update the doc and make a clear mention of this behaviour so that the user has no surprise (unwanted new resources created).

Reply:
That's a good point. Let me get back to you on this once I have the next change in place. 
I see that I could actually keep this script as before and instead pass a randomized EFS filename as a parameter so the test runs dont interfere with each other.

Reply:
I reverted the changes I made to your script and have used the script directly in the test scripts.